nifi-nar-bundles/nifi-iotdb-bundle/nifi-iotdb-processors/src/main/java/org/apache/nifi/processors/PutIoTDB.java

```diff
@@ -0,0 +1,286 @@
+                        processSession.transfer(flowFile, REL_FAILURE);
+                        return;
+                    }
+                }
+
+                long timestamp;
+                if (needInitFormatter) {
+                    timestamp = timeFormatter.parse((String) values[0]).getTime();
+                } else {
+                    timestamp = (Long) values[0];
+                }
+
+                boolean flag = false;
+
+                for (Map.Entry<String, Tablet> entry : tablets.entrySet()) {
+                    String device = entry.getKey();
+                    Tablet tablet = entry.getValue();
+                    int rowIndex = tablet.rowSize++;
+
+                    tablet.addTimestamp(rowIndex, timestamp);
+                    List<MeasurementSchema> measurements = tablet.getSchemas();
+                    for (MeasurementSchema measurement : measurements) {
+                        String tsName =
+                                new StringBuilder()
+                                        .append(device)
+                                        .append(".")
+                                        .append(measurement.getMeasurementId())
+                                        .toString();
+                        int valueIndex = fieldNames.indexOf(tsName) + 1;
+                        Object value;
+                        try {
+                            TSDataType type = measurement.getType();
+                            value = convertType(values[valueIndex], type);
+                        } catch (Exception e) {
+                            value = null;
+                        }
+                        tablet.addValue(measurement.getMeasurementId(), rowIndex, value);
+                    }
+                    flag = tablet.rowSize == tablet.getMaxRowNumber();
+                }
+                if (flag) {
+                    if (aligned) {
+                        session.get().insertAlignedTablets(tablets);
+                    } else {
+                        session.get().insertTablets(tablets);
+                    }
+                    tablets.values().forEach(tablet -> tablet.reset());
+                }
+            }
+
+            AtomicBoolean hasRest = new AtomicBoolean(false);
+            tablets.forEach(
+                    (device, tablet) -> {
+                        if (hasRest.get() == false && tablet.rowSize != 0) {
+                            hasRest.set(true);
+                        }
+                    });
+            if (hasRest.get()) {
+                if (aligned) {
+                    session.get().insertAlignedTablets(tablets);
+                } else {
+                    session.get().insertTablets(tablets);
+                }
+            }
+
+            inputStream.close();
+            recordReader.close();
+            processSession.transfer(flowFile, REL_SUCCESS);
+        } catch (Exception e) {
+            getLogger().error(e.getMessage());
```

Comment:
The exception should be logged with a message to ensure the full stack trace is available:
```suggestion
            getLogger().error("Processing failed {}", flowFile, e);
```